File: Praktikum2/C++/Main.cpp
#include <bits/stdc++.h>
using namespace std;

#include "Memory.cpp"

int main(int argc, char const *argv[])
{
    Memory comp[100];
    bool check = true;
    int i = 0;
    while (check)
    {
        string brand, model, frequency, supportsCuda;
        int price, idProduct, memorySize;

        cout << "Input: " << endl;
        cout << "Id Product: ";
        cin >> idProduct;
        cout << "Price: ";
        cin >> price;
        cout << "Brand: ";
        cin >> brand;
        cout << "Model: ";
        cin >> model;
        cout << "Frequency: ";
        cin >> frequency;
        cout << "Memory Size: ";
        cin >> memorySize;
        cout << "Supports Cuda: ";
        cin >> supportsCuda;

        comp[i].setIdProduct(idProduct);
        comp[i].setPrice(price);
        comp[i].setBrand(brand);
        comp[i].setModel(model);
        comp[i].setFrequency(frequency);
        comp[i].setMemorySize(memorySize);
        comp[i].setSupportsCuda(supportsCuda);
        char next;
        cout << "Lanjut memasukan data (Y/N): ";
        cin >> next;
        if (next == 'y' || next == 'Y')
        {
            check = true;
        }
        else
        {
            check = false;
        }
        i++;
    }

    cout << "\n";
    for (int j = 0; j < i; j++)
    {
        cout << "Output ke-" << j + 1 << endl;
        comp[j].printProduct();
        comp[j].printHardware();
        comp[j].printMemory();
        cout << endl;
    }
    return 0;
}

File: Praktikum2/Java/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Memory comp1 = new Memory();
        String brand = "", model = "", frequency = "", supportsCuda = "";
        int price = 0, idProduct = 0, memorySize = 0;

        System.out.println("Input: ");
        try {
            System.out.print("Id Product: ");
            idProduct = sc.nextInt();
        } catch (Exception e) {
        }
        try {
            System.out.print("Price: ");
            price = sc.nextInt();
        } catch (Exception e) {
        }
        try {
            System.out.print("Brand: ");
            brand = sc.next();
        } catch (Exception e) {
        }
        try {
            System.out.print("Model: ");
            model = sc.next();
        } catch (Exception e) {
        }
        try {
            System.out.print("Frequency: ");
            frequency = sc.next();
        } catch (Exception e) {
        }
        try {
            System.out.print("Memory Size: ");
            memorySize = sc.nextInt();
        } catch (Exception e) {
        }
        try {
            System.out.print("Supports Cuda: ");
            supportsCuda = sc.next();
        } catch (Exception e) {
        }

        comp1.setIdProduct(idProduct);
        comp1.setPrice(price);
        comp1.setBrand(brand);
        comp1.setModel(model);
        comp1.setFrequency(frequency);
        comp1.setMemorySize(memorySize);
        comp1.setSupportsCuda(supportsCuda);

        System.out.println("\nOutput: ");
        comp1.printProduct();
        comp1.printHardware();
        comp1.printMemory();
    }
}